### PR TITLE
Bump LightGBM version

### PR DESCRIPTION
**Pull request by shenggwang**
Bump LightGBM from 2.3.0 to 2.3.1.
Improve the way to define the LightGBM version.

### CODE REVIEW

File: openml-lightgbm/make.sh [hunk 1]
```diff
@@ -16,15 +16,28 @@
...
+    echo "Renaming the folder."
+    rm -rf lightgbmlib_build
+  fi
+fi
+
+# Make LightGBM if it doesn't exist
+DIR=lightgbmlib_build
+if [ ! -d "$DIR" ]; then
+  echo "Entering the folder."
   cd make-lightgbm || return
```
Comment (AlbertoEAF) [inline openml-lightgbm/make.sh:35]: Remove all return, they are invalid.

Use `set -e` instead to abort at any error.

File: openml-lightgbm/make.sh [hunk 2]
```diff
@@ -16,15 +16,28 @@
...
+# Compare version, when different remove the build folder
+FILE=lightgbmlib_build/__version__
+if [ -f "$FILE" ]; then
+  OLD_VERSION=$(head -n 1 "$FILE")
+  if [ "$VERSION" != "$OLD_VERSION" ]; then
+    echo "Renaming the folder."
+    rm -rf lightgbmlib_build
+  fi
+fi
+
```
Comment (AlbertoEAF) [inline openml-lightgbm/make.sh:30]: Add:

```suggestion
set -e
```

so any error aborts the script from here on.

File: openml-lightgbm/pom.xml [hunk 3]
```diff
@@ -24,6 +24,7 @@
 
     <properties>
         <lightgbmlib.version>2.3.150</lightgbmlib.version>
+        <make.lightgbmlib.version>2.3.1</make.lightgbmlib.version>
```
Comment (AlbertoEAF) [inline openml-lightgbm/pom.xml:27]: Use explicit versioning so we can replace by any commit in the future.
```suggestion
        <make.lightgbmlib.version>v2.3.1</make.lightgbmlib.version>
```

File: openml-lightgbm/make.sh [hunk 4]
```diff
@@ -16,15 +16,28 @@
 #
 # @author Sheng Wang (sheng.wang@feedzai.com)
 
-FILE=lightgbmlib_build
-if [ ! -d "$FILE" ]; then
-  echo "entering the folder."
+VERSION=v"$1"
```
[the remark this answers is not in the document]
Comment (shenggwang) [reply openml-lightgbm/make.sh:19]: Right, my bad, I just confirmed that the version is v2.3.1 and not only 2.3.1

File: openml-lightgbm/make.sh (hunk 2 above)
Comment (shenggwang) [reply openml-lightgbm/make.sh:30]: It indeed is more recommendable than `return`
[Source](https://stackoverflow.com/a/19622569)

**Comment by JPDSousa** [comment]
Tackles #27

**Comment by shenggwang** [comment]
I don't know why my second commit isn't here, but it is in the branch `ft-sw-PULSEDEV-31197-bump-lightgbm-version`.
Github was buggy in the morning, I hope that with merge it will correct this.

**Pull request merged**